components/suggestion/modal_suggestion_list.jsx

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import SuggestionList from 'components/suggestion/suggestion_list.jsx';
+import {getClosestParent} from 'utils/utils.jsx';
+
+export default class ModalSuggestionList extends React.PureComponent {
+    static propTypes = {
+        location: PropTypes.string.isRequired,
+        calculateInputRect: PropTypes.func.isRequired,
+        onLoseVisibility: PropTypes.func.isRequired,
+    }
+
+    latestHeight = 0;
+    scrollRegistered = false;
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            scroll: 0,
+            modalBounds: {top: 0, bottom: 0},
+            inputBounds: {top: 0, bottom: 0, width: 0},
+            location: props.location,
+        };
+
+        this.container = React.createRef();
+    }
+
+    tryRegisterScroll = () => {
+        if (this.container.current) {
+            const modalBodyContainer = getClosestParent(ReactDOM.findDOMNode(this.container.current), '.modal-body');
+            modalBodyContainer.addEventListener('scroll', this.onModalScroll);
+            this.scrollRegistered = true;
+        }
+    }
+
+    onModalScroll = (e) => {
+        if (this.state.scroll !== e.target.scrollTop &&
+            this.latestHeight !== 0) {
+            this.setState({scroll: e.target.scrollTop});
+        }
+    }
+
+    componentDidMount() {
+        this.tryRegisterScroll();
+        window.addEventListener('resize', this.updateModalBounds);
+    }
+
+    componentWillUnmount() {
+        if (this.container.current && this.scrollRegistered) {
+            const modalBodyContainer = getClosestParent(ReactDOM.findDOMNode(this.container.current), '.modal-body');
+            modalBodyContainer.removeEventListener('scroll', this.onModalScroll);
+        }
+        window.removeEventListener('resize', this.updateModalBounds);
+    }
+
+    componentDidUpdate() {
+        if (!this.scrollRegistered) {
+            this.tryRegisterScroll();
+        }
+
+        if (this.getChildHeight() === 0) {
+            return;
+        }
+
+        const newInputBounds = this.updateInputBounds();
+        this.updateLocation(newInputBounds);
+
+        if (this.container.current && newInputBounds) {
+            const modalBodyRect = getClosestParent(ReactDOM.findDOMNode(this.container.current), '.modal-body').getBoundingClientRect();
+            if ((newInputBounds.bottom < modalBodyRect.top) || (newInputBounds.top > modalBodyRect.bottom)) {
+                this.props.onLoseVisibility();
+                return;
+            }
+        }
+
+        this.updateModalBounds();
+    }
+
+    getChildHeight = () => {
+        if (!this.container.current) {
+            return 0;
+        }
+
+        const container = ReactDOM.findDOMNode(this.container.current);
```

Comment:
You can actually remove the `ReactDOM.findDOMNode` here. If a ref is on a regular HTML element, you'll already get the DOM node.